### PR TITLE
RE-1169 Add unit test for container slaves

### PR DESCRIPTION
**Pull request by hughsaunders**
RE-1169 requires upgrading the jenkins docker workflow plugin, to ensure
that the container standard slave type is still functioning after
this change and future changes, an additional unit test is added.

Issue: [RE-1169](https://rpc-openstack.atlassian.net/browse/RE-1169)

### CODE REVIEW

**Comment by hughsaunders** [comment]
recheck_unit_tests

**Comment by hughsaunders** [comment]
recheck_unit_tests

**Comment by hughsaunders** [comment]
recheck_xenial_lint_test recheck_xenial_functional_test

File: lint.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ install(){
     virtualenv $venv
   fi
   . $venv/bin/activate
-  pip install -c constraints.txt -r test-requirements.txt >/dev/null
+  pip install -c constraints.txt -r test-requirements.txt -r requirements.txt >/dev/null
```
Comment (mattt416) [inline lint.sh]: Hey @hughsaunders , why is this change necessary all of a sudden?

File: rpc_jobs/unit/single_use_slave.yml [hunk 2]
```diff
@@ -14,6 +14,10 @@
           REGIONS: "DFW ORD"
           FALLBACK_REGIONS: "IAD"
       - rpc_gating_params
+      - string:
+          name: SLAVE_TYPE
+          default: container
```
Comment (mattt416) [inline rpc_jobs/unit/single_use_slave.yml]: You could use the `standard_job_params` params here instead.  The default there is `instance`.

File: rpc_jobs/unit/re_unit_tests.yml [hunk 3]
```diff
@@ -49,15 +49,38 @@
...
               [
                 $class: 'StringParameterValue',
                 name: 'RPC_GATING_BRANCH',
                 value: env.RPC_GATING_BRANCH
+              ],
+              [
+                $class: 'StringParameterValue',
+                name: 'SLAVE_TYPE',
+                value: 'instance'
+              ]
```
Comment (mattt416) [inline rpc_jobs/unit/re_unit_tests.yml]: Ideally we'd want to unset this and allow it to use the default ("instance"), but we'd then need to add something into rpc_jobs/unit/single_use_slave.yml to actually validate we've gotten an instance, and not a container.  Know of any straight-forward way to do that?

File: lint.sh (hunk 1 above)
Comment (hughsaunders) [reply lint.sh]: I think its usage of jmespath, I can't remember where it is though.

File: rpc_jobs/unit/single_use_slave.yml (hunk 2 above)
Comment (hughsaunders) [reply rpc_jobs/unit/single_use_slave.yml]: Ah good plan.

File: rpc_jobs/unit/re_unit_tests.yml (hunk 3 above)
Comment (hughsaunders) [reply rpc_jobs/unit/re_unit_tests.yml]: systemd-detect-virt?

Comment (mattt416) [reply rpc_jobs/unit/re_unit_tests.yml]: Brilliant :)

File: rpc_jobs/unit/single_use_slave.yml (hunk 2 above)
Comment (mattt416) [reply rpc_jobs/unit/single_use_slave.yml]: Adjusted this.

File: rpc_jobs/unit/re_unit_tests.yml (hunk 3 above)
Comment (mattt416) [reply rpc_jobs/unit/re_unit_tests.yml]: Updated to do some actual assertions against the output of this.

File: lint.sh (hunk 1 above)
Comment (git-harry) [reply lint.sh]: So what is the distinction between requirements and test-requirements?

Comment (mattt416) [reply lint.sh]: Deferring to @hughsaunders on this one, since he made that specific change.

Comment (mattt416) [reply lint.sh]: Note this same change is in https://github.com/rcbops/rpc-gating/pull/629/ ... so I don't think this is needed by this specific PR?

Comment (git-harry) [reply lint.sh]: Why is it here then? Looking at the lint jobs, they are not failing.

**Comment by mattt416** [comment]
recheck_cit_unit_tests

**Comment by mattt416** [comment]
recheck_cit_xenial_functional_test
